### PR TITLE
merge main to PE

### PR DESCRIPTION
**Pull request by jslhcl**
### Description
merge main to PE

### Motivation and Context
merge main to PE

### CODE REVIEW

File: onnxruntime/contrib_ops/rocm/bert/layer_norm.cuh [hunk 1]
```diff
@@ -109,6 +109,49 @@ __device__ inline void LayerNorm(
   }
 }
 
+template <typename T, int TPB, int ILP>
+__device__ inline void LayerNormVec(
```
Comment (souptc) [inline onnxruntime/contrib_ops/rocm/bert/layer_norm.cuh:113]: what is this? we write a new kernel?

File: onnxruntime/contrib_ops/rocm/bert/skip_layer_norm_tunable_op.h [hunk 2]
```diff
@@ -54,30 +56,50 @@ Status SkipLayerNormSmallOp(const SkipLayerNormParams<T>* params) {
...
 }
 
-#define ADD_OP(threads_per_block)                                         \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 1>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 2>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 4>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 8>); \
-  this->ops_.emplace_back(SkipLayerNormSmallOp<T, threads_per_block, 16>);
+template <typename T, int ThreadsPerBlock, int VecSize>
+Status SkipLayerNormRegularOp(const SkipLayerNormParams<T>* params) {
```
Comment (souptc) [inline onnxruntime/contrib_ops/rocm/bert/skip_layer_norm_tunable_op.h:60]: please be careful about rocm EP, our pipeline is not fixed for rocm EP yet, so you can't verify the build in pipeline. 
please make sure those files are identity to main branch as we haven't work on AMD gpu yet.

File: onnxruntime/python/tools/kernel_explorer/kernels/skip_layer_norm.cc [hunk 3]
```diff
@@ -38,6 +38,30 @@ class SkipLayerNormSmall : public IKernelExplorer {
   ParamsT params_{};
 };
 
+template <typename T, int ThreadsPerBlock, int VecSize>
+class SkipLayerNormRegular : public IKernelExplorer {
```
Comment (souptc) [inline onnxruntime/python/tools/kernel_explorer/kernels/skip_layer_norm.cc:42]: just double confirm, those are the same as main branch, right?

File: onnxruntime/contrib_ops/rocm/bert/layer_norm.cuh (hunk 1 above)
Comment (jslhcl) [reply onnxruntime/contrib_ops/rocm/bert/layer_norm.cuh:113]: This is from main branch.

File: onnxruntime/python/tools/kernel_explorer/kernels/skip_layer_norm.cc (hunk 3 above)
Comment (jslhcl) [reply onnxruntime/python/tools/kernel_explorer/kernels/skip_layer_norm.cc:42]: yes, they are the same as main branch

File: onnxruntime/contrib_ops/rocm/bert/skip_layer_norm_tunable_op.h (hunk 2 above)
Comment (cloudhan) [reply onnxruntime/contrib_ops/rocm/bert/skip_layer_norm_tunable_op.h:60]: @Peixuanzuo and I are actively working on it. The CI will soon be enabled for ROCm. Drop us a msg if you have any question.

**Pull request closed without merging**